필기 도구를 이용해 슬라이드에 원하는 필기를 기입할 수 있는 기능

always-awake opened the pull request:
# 필기 도구를 이용해 슬라이드에 원하는 필기를 기입할 수 있는 기능

## 해당 이슈 📎

- 필기 도구를 이용해 슬라이드에 원하는 필기를 기입할 수 있다. (#86)

## 변경 사항 🛠

<img width="1417" alt="스크린샷 2019-12-12 오전 10 58 25" src="https://user-images.githubusercontent.com/40539104/70676383-564e4700-1ccf-11ea-86a9-152ce93d9dd8.png">

## 테스트 ✨
> 없음

## 리뷰어 참고 사항 🙋‍♀️

> 없음

always-awake committed:
[ CHORE ] hooks 내 Reducer 로직을 reducers 폴더로 분리

Why
- hook과 reducer는 다른 로직이기 때문에 분리하여 관리해야 한다.

What
- reducers 폴더를 만들고, channels 폴더와 main 폴더로 구분하여 reducer를 분리하였다.

always-awake committed:
Merge branch 'develop' of https://github.com/connect-foundation/2019-02 into feature/issue86-v3

always-awake committed:
[ MERGE ] develop -> feature/issue86-v3 충돌 해결

* develop -> feature/issue86-v3

always-awake committed:
[ FEATURE ] 슬라이드에 필기할 수 있는 기능 구현 (#86)

Why
- 스피커는 자신의 슬라이드 위에 필기를 할 수 있어야 한다.

What
- 필기도구 버튼을 클릭하면 툴바가 나타나도록 구현하였다.
- ToolBar 내에 PenTool을 클릭하면, 스피커는 슬라이드 위에 필기할 수 있다.

always-awake committed:
Merge branch 'develop' into feature/issue86-v3

aereeeee commented on web/src/components/channel/Slide/SlideViewer/MainSlide/MainSlide.jsx:
```diff
@@ -4,46 +4,62 @@ import React, {
-  const resizeCanvas = (fitHeight, width, height) => {
-    const canvasWidth = fitHeight ? height * slideRatio : width;
-    const canvasHeight = fitHeight ? height : width / slideRatio;
+  const resizeCanvas = (fitHeight, wrapperWidth, wrapperHeight) => {
```
resize관련 함수를 util로 뺄수 있을것 같아요!

devjaewon replied:
저도 동의합니다~!

happydhKim commented on web/src/components/channel/Slide/SlideViewer/Indicator/style.jsx:
```diff
@@ -22,6 +22,7 @@ export default {
         opacity: 1;
       }
     }
+    border: 1px solid red;
```
${px} 를!!!😥

always-awake replied:
테스트 중에 사용했던 스타일인데 삭제하는 걸 까먹었었네요 삭제하겠습니다 🙆‍♀️

aereeeee commented on web/src/components/channel/Slide/SlideViewer/MainSlide/SlideCanvas/SlideCanvas.jsx:
```diff
@@ -0,0 +1,83 @@
+    ctx.lineTo(event.offsetX, event.offsetY);
+    ctx.stroke();
+
+    setPosition(event.offsetX, event.offsetY);
```
캔버스 컨텍스트 생성하는 부분은 hook으로, 라인그리는 부분은 유틸로 빼면 좋을 것 같아요! 
나중에 시간이 되면 스타일 등을 설정하는 인터페이스도 만들고 라인그리기는 메소드로 만들어서 그리기툴 모듈을 만들면 환상적일 것 같습니다!

happydhKim commented on web/src/components/channel/Slide/SlideViewer/MainSlide/MainSlide.jsx:
```diff
@@ -4,46 +4,62 @@ import React, {
+            payload: prevCanvasUrl,
+          });
+        }
+        resizeCanvas(fitHeight, wrapperWidth, wrapperHeight);
```
함수를 쪼갤 수 있을 거 같습니다. resizeCanvas는 오브젝트 객체로 파라미터를 하나로 줄일다면, 더 깔끔해 지지 않을까요?🤔
1. resizeCanvas의 파라미터들을 구하는 함수를 따로 만들 수 있을거 같아요!
2. 파라미터를 하나로 만들 수 있을거 같아요!

어떻게 생각하시나여??🤠

aereeeee commented on web/src/reducers/channel.js:
```diff
@@ -11,6 +15,26 @@ const channelReducer = (state, action) => {
         ...state,
         isChat: action.payload.isChat,
       };
+    case 'toolBarActive':
```
액션 이름은 대문자 스네이크 케이스로 컨벤션을 맞추면 어떨까요? 다른 변수들과 구분되어서 더 좋은 것 같습니다!

always-awake replied:
넵 좋다고 생각됩니당 👌

happydhKim commented on web/src/components/channel/Slide/SlideViewer/MainSlide/SlideCanvas/SlideCanvas.jsx:
```diff
@@ -0,0 +1,83 @@
+import S from './style';
+import { useChannelSelector } from '@/hooks';
+
+const SlideCanvas = (_, ref) => {
```
SlideCanvas 컴포넌트의 연산하는 부분을 따로 나눠서 관리를 했으면 좋겠어요!! 🤠

devjaewon replied:
canvas high-level 모듈을 따로 만들어봐도 좋을 것 같아요!👍

aereeeee reviewed (approved):
고생많으십니다 멋진 캔버스 기능이 곧 완성되면 좋겟네요~!🥰

happydhKim commented on web/src/components/channel/Slide/SlideViewer/MainSlide/style.jsx:
```diff
@@ -25,9 +25,10 @@ export default {
     top: 50%;
     left: 50%;
     transform: translate(-50%, -50%);
+    border: 1px solid blue;
```
px이 힝.......

always-awake replied:
역시 삭제하겠습니다~

happydhKim commented on web/src/components/channel/Slide/SlideViewer/MainSlide/SlideCanvas/style.jsx:
```diff
@@ -0,0 +1,14 @@
+
+export default {
+  Canvas: styled.canvas`
+
```
공백은 절 위한 공간인가여??🥺

always-awake replied:
수정완료!

happydhKim commented on web/src/components/channel/Slide/SlideViewer/MainSlide/SlideCanvas/style.jsx:
```diff
@@ -0,0 +1,14 @@
+import styled from 'styled-components';
+// import { px } from '@/styles/themeUtil';
```
의미 없는 주석은 지워버리죠!!😀

always-awake replied:
수정 완료!

happydhKim reviewed (approved):
와우 캔버스가 점점 완벽해지는거 같아요 정말 고생하셨어요

팀을 위한 노력 !! 감사합니다!!😎

devjaewon reviewed (approved):
어려운 기능을 만드시느라 고생하시네요 🥺
마지막까지 화이팅입니다!!

always-awake committed:
[ FIX ] 리뷰 사항 반영

Why
- 리뷰어들의 리뷰사항을 반영해야 한다.

What
- reducer에 action명을 대문자로 수정하였다.
- 불필요한 코드를 삭제해주었다.

always-awake committed:
Merge branch 'feature/issue86-v3' of https://github.com/connect-foundation/2019-02 into feature/issue86-v3
